src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/PosixSignalRegistration.Windows.cs

```diff
@@ -1,18 +1,134 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System;
+using System.Collections.Generic;
+using System.IO;
 
 namespace System.Runtime.InteropServices
 {
-    public sealed class PosixSignalRegistration : IDisposable
+    public sealed partial class PosixSignalRegistration
     {
-        private PosixSignalRegistration() { }
+        private static readonly Interop.Kernel32.ConsoleCtrlHandlerRoutine s_handlerRoutine = HandlerRoutine;
```

Comment:
You can switch this to use function pointers while you are on it. It was not easy to use function pointers in the old console implementation since the target was an instance method.

Reply:
Will do